### PR TITLE
Fix a crash when disposing tabs

### PR DESCRIPTION
**Pull request by Hixie**
...and some code cleanup.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @goderbauer

File: packages/flutter/test/material/tabs_test.dart [hunk 1]
```diff
@@ -2670,6 +2670,13 @@ void main() {
     final PageView pageView = tester.widget<PageView>(find.byType(PageView));
     expect(pageView.physics.toString().contains('ClampingScrollPhysics'), isFalse);
   });
+
+  testWidgets('Crash on dispose', (WidgetTester tester) async {
+    await tester.pumpWidget(Padding(padding: const EdgeInsets.only(right: 200.0), child: TabBarDemo()));
+    await tester.tap(find.byIcon(Icons.directions_bike));
+    // There was a time where this would not throw an exception
```
Comment (goderbauer) [inline packages/flutter/test/material/tabs_test.dart:2677]: remove the "not"? As in it used to throw and now it doesn't anymore?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/test/material/tabs_test.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/test/material/tabs_test.dart:2677]: oops yeah

**Pull request merged**